### ISSUE TITLE
Size of parameter dialogs with replaceable components

### ISSUE THREAD

**Issue by abuntrock** [OpenModelica/OpenModelica#10395]
### Description

Parameter dialogues that include replaceable component drop downs are very wide and can even result in the need of horizontal scrollbars. This makes them harder to use in our opinion.

### Steps to Reproduce

1. Load the BRSL 2.2.1.
2. Open a new Modelica class.
3. Instantiate `Rexroth_Generic.BRSL_World` and then `Rexroth_Cylinders.MillTypeCylinders.CD.CDH1`.
4. Open the parameter dialogue of the cylinder. It looks as shown in the first two screenshots. For component `cylinder` the complete replaceable component declaration including all modifiers together with a lot of white space on the right is shown in the drop-down menu. As a result, one cannot see the name and the unit of parameters (e.g. `stroke`) at the same time. You also have to scroll to the right to expand the drop-down menu of `cylinder` and to see the new edit button for the redeclared class. Since you are not able to reduce the width of the parameter dialogue, this makes parametrization unnecessarily hard.
5. Select any choice for parameter `cylinder.`
6. Close and reopen the parameter dialogue of the instantiated `CDH1`. As shown in the third screenshot the width of the `cylinder` parameter is reduced, which is great. However, there is a lot of unnecessary white space for all parameters, which are in another group than `cylinder`. 

### Expected Behavior

The parameter dialogue always shows the name and the unit of all parameters without the need to scroll horizontally.
From our perspective the complete replaceable declaration including all modifiers does not need to be seen with the default width of the parameter dialogue. Our proposal would be something like the Dymola solution shown in the last screenshot. The drop-down menu does not show the complete replaceable declaration (with all modifiers) with the default dialogue width.
In addition, the dialogue should be rescaled better after selecting a replaceable choice and reopening the dialogue.

### Screenshots

@niklwors will add screenshots in the comment since I am not allowed to upload them from my work laptop.

### Version and OS

 - OpenModelica Version: 1.21.0-dev-342-gcfdfe6e76e (64-bit) with instance API
 - OS: Windows 10, 64 bit
 - MSL 4.0.0, BRSL2.2.1
 
Thanks for looking into this.
Aaron Buntrock - Bosch rexroth

**Comment by niklwors** [comment]
picture 1:
![image001](https://user-images.githubusercontent.com/12269776/224723649-70ea105c-5927-4317-9047-c5117f271613.png)

picture 2:
![image002](https://user-images.githubusercontent.com/12269776/224723667-78eefb79-8730-44bf-bdf2-32854ff0f8cd.png)

picture 3:
![image003](https://user-images.githubusercontent.com/12269776/224723689-201c0e72-13e3-438c-8cc9-7d34892baa7f.png)

picture 4:
![image004](https://user-images.githubusercontent.com/12269776/224723724-3b4dff8c-c199-4452-bed7-d3cee5a37e09.png)

**Comment by niklwors** [comment]
The parameter dialog looks much better now. Some fields are still a little bit too large.
<img width="1891" alt="dialog" src="https://github.com/OpenModelica/OpenModelica/assets/12269776/a67e62c5-7342-467a-aa3f-8233221fbbda">

**Comment by adeas31** [comment]
Yes this is after #10666. I actually made the value controls i.e., textboxes and comboboxes to take as much of the screen space available if needed. If we want to further improve it then I have to change this behavior so the controls are not so hungry for the space. But then we might see small boxes with huge values and the values are not fully visible.

**Comment by abuntrock** [comment]
With very large arrays, whose value would require a scrollbar, the scrollbar is not there anymore fr me with today's nightly. The small scrollbar amount only appears when there are replaceable components/classes with a dropdown. Maybe this might help.

**Comment by adeas31** [comment]
@abuntrock can you provide a test model?

**Comment by abuntrock** [comment]
@adeas31 During the creation of a minimum example I found out that the scrollbar appeared for the last picture @niklwors posted because the description string of `axialPistonUnit` was too long. We think (as we expect it is implemented) that the description string has to be always shown completely. 
It seems that the description string can only "take away" a certain amount of horizontal space from the parameter value before a scrollbar is created. I assume this is hard to fix and not clear for all cases, so I would be OK with the current solution regarding this. 

However, during testing I recognized another issue which can also be seen in @niklwors last picture (there for the parameter `fluidIndex`). To hightlight this I, I created the following minimum example:

```modelica
model LargeArray

 parameter Integer Integer1 = 1 "Some description string" annotation(Dialog(group = "NotLargeTable"));
 parameter Integer Integer2 = 2 "Some description string" annotation(Dialog(group = "LargeTable"));
 parameter Real[:,:] largeArray = [0,0,5,7,10,15,20,30,40,50,100,200,400; 0,0,0,0,0,0,0,0,0,0,0,0,0;
        0.0016,0,-0.00689,-0.00659,-0.00616,-0.00543,-0.00469,-0.00323,-0.00177,
        -0.000304,0.00703,0.0217,0.0514; 0.0032,0,-0.00656,-0.00626,-0.00582,-0.00509,
        -0.00436,-0.0029,-0.00144,2.74e-05,0.00736,0.0221,0.0518; 0.0048,0,-0.00615,
        -0.00586,-0.00542,-0.00469,-0.00396,-0.0025,-0.00103,0.000432,0.00777,
        0.0225,0.0522; 0.0064,0,-0.00566,-0.00537,-0.00493,-0.0042,-0.00347,-0.00201,
        -0.000545,0.000919,0.00825,0.023,0.0527; 0.008,0,-0.0051,-0.0048,-0.00437,
        -0.00364,-0.0029,-0.00144,2.18e-05,0.00149,0.00882,0.0236,0.0533;
        0.0096,0,-0.00446,-0.00417,-0.00373,-0.003,-0.00227,-0.000807,0.000656,
        0.00212,0.00946,0.0242,0.0539; 0.0112,0,-0.00379,-0.0035,-0.00306,-0.00233,
        -0.0016,-0.000134,0.00133,0.00279,0.0101,0.0249,0.0546; 0.0128,0,-0.00311,
        -0.00282,-0.00238,-0.00165,-0.000919,0.000544,0.00201,0.00347,0.0108,
        0.0255,0.0553; 0.0144,0,-0.00247,-0.00217,-0.00174,-0.00101,-0.000274,
        0.00119,0.00265,0.00412,0.0115,0.0262,0.0559; 0.016,0,-0.00188,-0.00159,
        -0.00115,-0.00042,0.000311,0.00177,0.00324,0.0047,0.012,0.0268,0.0565;
        0.0176,0,-0.00137,-0.00108,-0.000641,8.99e-05,0.000821,0.00228,0.00375,
        0.00521,0.0125,0.0273,0.057; 0.0192,0,-0.00094,-0.000648,-0.00021,
        0.000521,0.00125,0.00271,0.00418,0.00564,0.013,0.0277,0.0574; 0.0208,0,
        -0.000585,-0.000293,0.000146,0.000876,0.00161,0.00307,0.00453,0.006,
        0.0133,0.0281,0.0578; 0.0224,0,-0.000298,-5.23e-06,0.000433,0.00116,
        0.00189,0.00336,0.00482,0.00629,0.0136,0.0284,0.0581; 0.024,0,-6.82e-05,
        0.000224,0.000662,0.00139,0.00212,0.00359,0.00505,0.00651,0.0138,0.0286,
        0.0583; 0.0256,0,0.000113,0.000405,0.000843,0.00157,0.0023,0.00377,
        0.00523,0.0067,0.014,0.0288,0.0585; 0.0272,0,0.000254,0.000546,0.000985,
        0.00172,0.00245,0.00391,0.00537,0.00684,0.0142,0.0289,0.0586; 0.0288,0,
        0.000364,0.000656,0.00109,0.00183,0.00256,0.00402,0.00548,0.00695,
        0.0143,0.029,0.0587; 0.0304,0,0.000449,0.000742,0.00118,0.00191,0.00264,
        0.0041,0.00557,0.00703,0.0144,0.0291,0.0588; 0.032,0,0.000515,0.000807,
        0.00125,0.00198,0.00271,0.00417,0.00563,0.0071,0.0144,0.0292,0.0589] "This is an unnecessarily long description string, which has no real meaning and results in a scrollbar at the bottom" annotation(Dialog(group = "LargeTable"));

end LargeArray;
```

I would expect that `Integer1` and `Integer2` look the same in the parameter dialog but `Integer1` gets a lot more unnecessary white space before the description string because it is not in the same parameter group as the large array. (See pictures from @niklwors below). As a result, I have the unnecessarily scroll to the right to see the description string for `Integer1`. I assume to Solution to this issue would be that maximum width of the parameter value is applied across all parameter groups.

**Comment by bilderbuchi** [comment]
I think it should be possible to tweak the [Qt SizePolicy](https://doc.qt.io/qt-5/qsizepolicy.html#Policy-enum) settings for those widgets to achieve a desired scaling behaviour, but I'm not sure if this is how OMEdit is using Qt.

**Comment by niklwors** [comment]
Here are the pictures from @abuntrock 

![image001](https://github.com/OpenModelica/OpenModelica/assets/12269776/002696d3-bfe9-47b9-bb76-aee81c288bb6)


![image002](https://github.com/OpenModelica/OpenModelica/assets/12269776/28d0d207-785c-436f-94fa-01e27df931bc)

**Comment by adeas31** [comment]
> [@adeas31](https://github.com/adeas31) During the creation of a minimum example I found out that the scrollbar appeared for the last picture [@niklwors](https://github.com/niklwors) posted because the description string of `axialPistonUnit` was too long. We think (as we expect it is implemented) that the description string has to be always shown completely. It seems that the description string can only "take away" a certain amount of horizontal space from the parameter value before a scrollbar is created. I assume this is hard to fix and not clear for all cases, so I would be OK with the current solution regarding this.
> 
> However, during testing I recognized another issue which can also be seen in [@niklwors](https://github.com/niklwors) last picture (there for the parameter `fluidIndex`). To hightlight this I, I created the following minimum example:
> 
> model LargeArray
> 
>  parameter Integer Integer1 = 1 "Some description string" annotation(Dialog(group = "NotLargeTable"));
>  parameter Integer Integer2 = 2 "Some description string" annotation(Dialog(group = "LargeTable"));
>  parameter Real[:,:] largeArray = [0,0,5,7,10,15,20,30,40,50,100,200,400; 0,0,0,0,0,0,0,0,0,0,0,0,0;
>         0.0016,0,-0.00689,-0.00659,-0.00616,-0.00543,-0.00469,-0.00323,-0.00177,
>         -0.000304,0.00703,0.0217,0.0514; 0.0032,0,-0.00656,-0.00626,-0.00582,-0.00509,
>         -0.00436,-0.0029,-0.00144,2.74e-05,0.00736,0.0221,0.0518; 0.0048,0,-0.00615,
>         -0.00586,-0.00542,-0.00469,-0.00396,-0.0025,-0.00103,0.000432,0.00777,
>         0.0225,0.0522; 0.0064,0,-0.00566,-0.00537,-0.00493,-0.0042,-0.00347,-0.00201,
>         -0.000545,0.000919,0.00825,0.023,0.0527; 0.008,0,-0.0051,-0.0048,-0.00437,
>         -0.00364,-0.0029,-0.00144,2.18e-05,0.00149,0.00882,0.0236,0.0533;
>         0.0096,0,-0.00446,-0.00417,-0.00373,-0.003,-0.00227,-0.000807,0.000656,
>         0.00212,0.00946,0.0242,0.0539; 0.0112,0,-0.00379,-0.0035,-0.00306,-0.00233,
>         -0.0016,-0.000134,0.00133,0.00279,0.0101,0.0249,0.0546; 0.0128,0,-0.00311,
>         -0.00282,-0.00238,-0.00165,-0.000919,0.000544,0.00201,0.00347,0.0108,
>         0.0255,0.0553; 0.0144,0,-0.00247,-0.00217,-0.00174,-0.00101,-0.000274,
>         0.00119,0.00265,0.00412,0.0115,0.0262,0.0559; 0.016,0,-0.00188,-0.00159,
>         -0.00115,-0.00042,0.000311,0.00177,0.00324,0.0047,0.012,0.0268,0.0565;
>         0.0176,0,-0.00137,-0.00108,-0.000641,8.99e-05,0.000821,0.00228,0.00375,
>         0.00521,0.0125,0.0273,0.057; 0.0192,0,-0.00094,-0.000648,-0.00021,
>         0.000521,0.00125,0.00271,0.00418,0.00564,0.013,0.0277,0.0574; 0.0208,0,
>         -0.000585,-0.000293,0.000146,0.000876,0.00161,0.00307,0.00453,0.006,
>         0.0133,0.0281,0.0578; 0.0224,0,-0.000298,-5.23e-06,0.000433,0.00116,
>         0.00189,0.00336,0.00482,0.00629,0.0136,0.0284,0.0581; 0.024,0,-6.82e-05,
>         0.000224,0.000662,0.00139,0.00212,0.00359,0.00505,0.00651,0.0138,0.0286,
>         0.0583; 0.0256,0,0.000113,0.000405,0.000843,0.00157,0.0023,0.00377,
>         0.00523,0.0067,0.014,0.0288,0.0585; 0.0272,0,0.000254,0.000546,0.000985,
>         0.00172,0.00245,0.00391,0.00537,0.00684,0.0142,0.0289,0.0586; 0.0288,0,
>         0.000364,0.000656,0.00109,0.00183,0.00256,0.00402,0.00548,0.00695,
>         0.0143,0.029,0.0587; 0.0304,0,0.000449,0.000742,0.00118,0.00191,0.00264,
>         0.0041,0.00557,0.00703,0.0144,0.0291,0.0588; 0.032,0,0.000515,0.000807,
>         0.00125,0.00198,0.00271,0.00417,0.00563,0.0071,0.0144,0.0292,0.0589] "This is an unnecessarily long description string, which has no real meaning and results in a scrollbar at the bottom" annotation(Dialog(group = "LargeTable"));
> 
> end LargeArray;
> I would expect that `Integer1` and `Integer2` look the same in the parameter dialog but `Integer1` gets a lot more unnecessary white space before the description string because it is not in the same parameter group as the large array. (See pictures from [@niklwors](https://github.com/niklwors) below). As a result, I have to unnecessarily scroll to the right to see the description string for `Integer1`. I assume the solution to this issue would be that maximum width of the parameter value is applied across all parameter groups.

Yes they belong to different groups so we can't get same layout. Setting maximum width is also not a good idea.

With PR #14723 I changed that the comment on the right side is shown in ellipses instead of taking the full horizontal space. You can see the full comment in tooltip.

**Comment by casella** [comment]
@abuntrock can you please check with the latest nightly build? Thanks!

**Comment by abuntrock** [comment]
I found some time to test this with today's nightly and overall I am really impressed. The drop down menus of replaceables and input fields of arrays are now not that wide anymore. Those were the two most pressing issues from our side. 
So I think we can close this issue here and discuss remaining points in #11721

**Comment by casella** [comment]
@abuntrock good to hear!
